src/libstore/ssh.cc

```diff
@@ -152,13 +159,15 @@ Path SSHMaster::startMaster()
 
     out.writeSide = -1;
 
-    std::string reply;
-    try {
-        reply = readLine(out.readSide.get());
-    } catch (EndOfFile & e) { }
+    if (wasMasterRunning) {
```

Comment:
Shouldn't that be the other way around?

```suggestion
    if (!wasMasterRunning) {
```

(maybe we could even make this whole function bail out early if `isMasterRunning() == true`, but let's save that for another time)